.github/workflows/root-ci.yml

```
@@ -356,12 +356,22 @@ jobs:
             overrides: ["imt=Off", "LLVM_ENABLE_ASSERTIONS=On", "CMAKE_BUILD_TYPE=Debug"]
           - image: ubuntu2404
             overrides: ["LLVM_ENABLE_ASSERTIONS=On", "CMAKE_BUILD_TYPE=Debug"]
+          # Special builds
+          - image: alma9
+            is_special: true
+            property: march_native
+            overrides: ["CMAKE_CXX_FLAGS=-march=native", "CMAKE_C_FLAGS=-march=native"]
```

```suggestion
            overrides: ["CMAKE_CXX_FLAGS=-march=native", "CMAKE_C_FLAGS=-march=native","CCACHE=OFF"]
```
avoid risks of incompatibility of the instruction set.

---

Good suggestion. On Linux we do not have ccache yet in the CI.

---

Also, the CMake options are part of the path to the artifacts. The overrides are also part of these options and modify the hash: https://github.com/root-project/root/blob/master/.github/workflows/root-ci-config/build_root.py#L88 . For what I understand from the code, and I could be wrong or the code buggy, the artifacts *will* be different because of the different "-D's"

---

> the artifacts will be different because of the different "-D's"

Yes but 2 different micro-architecture might result is distinct (possibly incompatible code) for the same `CMAKE_C_FLAGS=-march=native`